src/unity/toolkits/tcmps/mps_graph_cnnmodule.mm

```diff
@@ -101,40 +120,43 @@ - (instancetype)initWithCommandBuffer:(id<MTLCommandBuffer>)commandBuffer {
     [lossState synchronizeOnCommandBuffer:cb];
   }
 
+  // Schedule copying of the output into a float_array promise.
+  // n.b. The block below must not capture `this`.
+  size_t loss_size = static_cast<size_t>(batch.lossState.count);
+  auto loss_promise = std::make_shared<std::promise<shared_float_array>>();
+  NSMutableArray<MPSImageBatch *> *recycled_inputs = recycled_inputs_;
```

Comment:
Is this local copy of the ref needed in case it changes? Maybe you can say something about why this is needed.

Reply:
Or is it simply so that the closure (or *block*, I guess they are called) doesn't have to wrap the entire `this`?